tests/scripts/thread-cert/Cert_5_3_06_RouterIdMask.py

```diff
@@ -89,64 +113,97 @@ def test(self):
-        self.assertFalse(command.check_id_set(msg, router1_id))
-        self.assertFalse(command.check_id_set(msg, router2_id))
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+
+        # Step 1: Ensure topology is formed correctly
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst64(LEADER).\
+            filter_mle_cmd(MLE_CHILD_ID_REQUEST).\
+            must_next()
+        pkts.filter_wpan_src64(LEADER).\
+            filter_wpan_dst64(ROUTER_1).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            must_next()
+        pkts.filter_wpan_src64(ROUTER_2).\
+            filter_wpan_dst64(ROUTER_1).\
+            filter_mle_cmd(MLE_CHILD_ID_REQUEST).\
+            must_next()
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst64(ROUTER_2).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            must_next()
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {1,2,1} == set(p.mle.tlv.route64.cost)).\
+            must_next()
+
+        # Step 4: The DUT’s routing cost to Router_2 MUST count to infinity
+        #         The DUT MUST remove Router_2 ID from its ID set
+        #         Verify route data has settled
+        _pkt = pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {1,0,1} == set(p.mle.tlv.route64.cost)).\
+            must_next()
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {1,1} == set(p.mle.tlv.route64.cost)).\
+            must_next()
+
+        # Step 5: Re-attach Router_2 to Router_1.
+        #         The DUT MUST reset the MLE Advertisement trickle timer and
+        #         send an Advertisement
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {1,2,1} == set(p.mle.tlv.route64.cost)).\
+            must_next()
+
+        # Step 6: The DUT’s routing cost to Router_1 MUST go directly to
+        #         infinity as there is no multi-hop cost for Router_1
+        #         The DUT MUST remove Router_1 & Router_2 IDs from its ID set
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {0, 0, 1} == set(p.mle.tlv.route64.cost)).\
+            must_next()
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {0, 1} == set(p.mle.tlv.route64.cost)).\
```

Comment:
`ROUTER_2` should be removed at the same time as `ROUTER_1` is removed.
Are there any chance for two route entries in Leader's MLE advertisement given the step 6 scenario?

Reply:
yes, from the packets, first remove ROUTER_2, then remove ROUTER_1 from the ROUTE64 entries